controllers/vsl.go

```diff
@@ -110,53 +110,37 @@ func (r *DPAReconciler) LabelVSLSecrets(log logr.Logger) (bool, error) {
 
 func (r *DPAReconciler) ValidateVolumeSnapshotLocations(dpa oadpv1alpha1.DataProtectionApplication) (bool, error) {
 	for i, vslSpec := range dpa.Spec.SnapshotLocations {
+		vslYAMLPath := fmt.Sprintf("spec.snapshotLocations[%v]", i)
+		veleroVSLYAMLPath := vslYAMLPath + ".velero"
+
 		if vslSpec.Velero == nil {
 			return false, errors.New("snapshotLocation velero configuration cannot be nil")
 		}
-		vsl := velerov1.VolumeSnapshotLocation{
-			ObjectMeta: metav1.ObjectMeta{
-				// TODO: Use a hash instead of i
-				Name:      fmt.Sprintf("%s-%d", r.NamespacedName.Name, i+1),
-				Namespace: r.NamespacedName.Namespace,
-			},
-			Spec: *vslSpec.Velero,
-		}
 
 		// check for valid provider
 		if vslSpec.Velero.Provider != AWSProvider && vslSpec.Velero.Provider != GCPProvider &&
 			vslSpec.Velero.Provider != Azure {
-			r.Log.Info("Non-supported provider specified, might be a misconfiguration")
-
-			r.EventRecorder.Event(&vsl,
-				corev1.EventTypeWarning,
-				"VSL provider is invalid",
-				fmt.Sprintf("VSL provider %s is invalid, might be a misconfiguration", vslSpec.Velero.Provider),
-			)
+			return false, fmt.Errorf("DPA %s.provider %s is invalid: only AWS, GCP and Azure are supported", veleroVSLYAMLPath, vslSpec.Velero.Provider)
 		}
 
 		//AWS
 		if vslSpec.Velero.Provider == AWSProvider {
 			//in AWS, region is a required field
 			if len(vslSpec.Velero.Config[AWSRegion]) == 0 {
-				return false, errors.New("region for AWS VSL is not configured, please ensure a region is configured")
+				return false, fmt.Errorf("region for AWS VSL in DPA %s.Config is not configured, please ensure a region is configured", veleroVSLYAMLPath)
 			}
 
 			// check for invalid config key
 			for key := range vslSpec.Velero.Config {
 				valid := validAWSKeys[key]
 				if !valid {
-					return false, fmt.Errorf("%s is not a valid AWS config value", key)
+					return false, fmt.Errorf("DPA %s.config key %s is not a valid AWS config key", veleroVSLYAMLPath, key)
 				}
 			}
 			//checking the aws plugin, if not present, throw warning message
 			if !containsPlugin(dpa.Spec.Configuration.Velero.DefaultPlugins, AWSProvider) {
-				r.Log.Info("VSL for AWS specified, but AWS plugin not present, might be a misconfiguration")
 
-				r.EventRecorder.Event(&vsl,
-					corev1.EventTypeWarning,
-					"VolumeSnapshotLocation is invalid",
-					fmt.Sprintf("could not validate vsl for AWS plugin on: %s/%s", vsl.Namespace, vsl.Name),
-				)
+				return false, fmt.Errorf("VSL for AWS specified in DPA %s, but AWS plugin not present, might be a misconfiguration", vslYAMLPath)
```

Comment:
better to use imperative form here, no?

would also add path to aws plugin config in DPA

Suggestion 

> to use VSL for AWS specified in DPA spec.snapshotLocations[index], AWS plugin must be present in spec.configuration.velero.defaultPlugins